app/views/deals/_form.html.erb

```diff
@@ -0,0 +1,28 @@
+<main class="transactions">
+  <h2 class="transactions-heading"> Add a New Transaction </h2>
+  <div class="transsactions-form-container">
+    <%= form_with(model: [@category, @deal], url: user_category_deals_path(current_user, @category), local: true, html: { id: 'new-transaction-save-btn', class: 'new-deal-form' }) do |form| %>
+      <div class="field name-field">
+        <%= form.label :name %>:
+        <%= form.text_field :name %>
+      </div>
+
+      <div class="field name-field">
+        <%= form.label :amount %>:
+        <%= form.number_field :amount %>
+      </div>
+
+      <div class="category-selector">
+        <p class="cat-label"><%= form.label :category_ids, "Categories" %></p>
+        <p>
+        <%= form.collection_select(:category_id, current_user.categories, :id, :name, { include_hidden: false }, { multiple: false }) %>
+        </p>
+      </div>
+
+    <% end %>
+  </div>
+  <%= link_to "Back to Transactions", user_category_path(current_user, @category) %>
+<div class="category-actions">
+<%= link_to "SAVE", 'javascript:void(0)', onclick: 'getElementById("new-transaction-save-btn").submit();', class: 'new-category-btn' %>
```

Comment:
- [ ] After creating a transaction successfully users are redirected to a blank page. Please make sure to fix this issue and after successfully creating transactions users should be redirected to the transactions page for that category as requested in the project requirements.

![image](https://github.com/ometman/budget-app/assets/83803414/ed8beabf-05df-437e-b579-be67daefe004)

Reply:
Thank you. Fixed